api/src/main/java/org/pmiops/workbench/api/UserController.java

```diff
@@ -160,7 +160,7 @@ private boolean isUserInRegisteredTier() {
         .anyMatch(
             tier ->
                 tier.getShortName().equals(AccessTierService.REGISTERED_TIER_SHORT_NAME)
-                    && fireCloudService.isUserMemberOfGroup(
+                    && fireCloudService.isUserMemberOfGroupWithCache(
```

Comment:
We may want to keep this one uncached: the only usage of this one is "share workspace" which should not be subject to overload.

OTOH, what's the quickest time a user might go from gains tier access -> share workspace?  Maybe a short potential cache delay wouldn't matter.

Aside: I'm planning to update this to multi-tier soon.  That's not likely to be relevant, just FYI

Reply:
To clarify, the cache added in this PR is request scoped - i.e. caching only happens **within a single request**, to avoid exactly these kinds of complications. Having a longer-lived cache in server memory, or split out into a separate memcache may make sense, but I wasn't willing to bite that off here as it needs to be done very carefully. 

Knowing that it is request scoped - do you still have concerns over this call site? It may still be safer to split this into an uncached version.

Reply:
oh that is really nice!  no concerns with any of these caches, then.